bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java

```diff
@@ -257,6 +306,8 @@ private void bindDefaultListener() {
      */
     @Override
     public void initialize() {
+        VolumioService volumioLocal = volumio;
```

Comment:
I think you forgot a null check here because the compiler prints:

`Warning:  /home/runner/work/openhab-addons/openhab-addons/bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:[331,17] Potential null pointer access: The variable volumioLocal may be null at this location`

Reply:
If i'm not mistaken, volumioLocal will always be null, as this method is only called when the class initializes and therefore will allways be null.

See also comment from @lsiepel

Reply:
That was because a preliminary return caused it never to instantiate. It still needs a check.

Reply:
Than I need help how can i remove this last compiler warning?? I tried a lot but always get this error...Can you help me?

Reply:
Assign it to the local variable and then the local variable to the field

```java 
Volumio service volumioLocal = new VolumioService(protocol, hostname, port, timeout);
volumio = volumioLocal;
```

Reply:
Getting this error

R] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,58] protocol cannot be resolved to a variable
[ERROR] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,68] hostname cannot be resolved to a variable
[ERROR] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,78] port cannot be resolved to a variable
[ERROR] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,84] timeout cannot be resolved to a variable

Reply:
The idea is to introduce the local var here:

https://github.com/openhab/openhab-addons/blob/11e954847f38d7987899e7d4faad82f1cde674f0/bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java#L327

Reply:
solved